pkg/e2e/state/alerts.go

```diff
@@ -49,60 +36,50 @@ func init() {
 
 var _ = ginkgo.Describe(clusterStateTestName, func() {
 	defer ginkgo.GinkgoRecover()
-	h := helper.New()
 
 	alertsTimeoutInSeconds := 900
 	ginkgo.It("should have no alerts", func() {
-		var results map[string][]byte
-		// setup runner
-		h.SetServiceAccount("system:serviceaccount:%s:cluster-admin")
-		r := h.RunnerWithNoCommand()
-
-		r.Name = "alerts"
-		serviceAccountDir := "/var/run/secrets/kubernetes.io/serviceaccount"
-
-		alertsCommand, err := h.ConvertTemplateToString(alertsCmdTpl, struct {
-			OutputDir string
-			Name      string
-			Server    string
-			CA        string
-			TokenFile string
-		}{
-			OutputDir: runner.DefaultRunner.OutputDir,
-			Name:      r.Name,
-			Server:    "https://kubernetes.default",
-			CA:        serviceAccountDir + "/ca.crt",
-			TokenFile: serviceAccountDir + "/token",
-		})
-		Expect(err).NotTo(HaveOccurred(), "failure creating templated command")
 
-		r.Cmd = alertsCommand
+		//Set up prometheus client
+		promClient, err := prometheus.CreateClient()
+
+		Expect(err).NotTo(HaveOccurred(), "error creating a prometheus client")
+		promAPI := promv1.NewAPI(promClient)
 
-		stopCh := make(chan struct{})
-		// run tests
-		err = r.Run(alertsTimeoutInSeconds, stopCh)
-		Expect(err).NotTo(HaveOccurred(), "error running alert gatherer")
+		var queryresult []byte
 
-		// get results
-		results, err = r.RetrieveResults()
+		// Query for alerts with a retry count of 40 and timeout of 20 minutes
+		err = wait.PollImmediate(30*time.Second, 20*time.Minute, func() (bool, error) {
+			query := fmt.Sprintf("ALERTS{alertstate!=\"pending\",alertname!=\"Watchdog\"}")
+			context, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+			defer cancel()
+			result, _, err := promAPI.Query(context, query, time.Now())
+			if err != nil {
+				return false, fmt.Errorf("Unable to query prom API")
+			}
+			queryresult, err = json.MarshalIndent(result, "", "  ")
```

Comment:
This seems a bit of a hack to interpret results. Instead, we should use Prometheus' native libraries/objects.

https://github.com/openshift/osde2e/blob/main/pkg/metrics/client.go#L62-L82 is a good example of retrieving a list of results and how to handle them. In this case, we could potentially see multiple alerts and, if we do, we should log all the alerts we've seen. 

This will impact the changes to `findCriticalAlerts` below, since you _should_ theoretically pass it an array of results (even if it's an empty array)